### ISSUE TITLE
Git Download Location regex matcher doesn't meet spec

### ISSUE THREAD

**Issue by ianling** [spdx/Spdx-Java-Library#53]
Spec: https://spdx.github.io/spdx-spec/3-package-information/#37-package-download-location

Code: https://github.com/spdx/Spdx-Java-Library/blob/c909ebcdbcca06e89a2e406362b261d5b7f1e729/src/main/java/org/spdx/library/SpdxConstants.java#L362

Here is a link to a bunch of test cases that should work according to the spec, but do not: https://regex101.com/r/mb4tWm/1

For reference in case that link dies, here's a summary of ones that do and do not work:

```
git://git.myproject.org/MyProject.git@master ✅
git://git.myproject.org/MyOrg/MyProject.git@master ✅
git+git@git.myproject.org:MyProject ✅
git+git@git.myproject.org:MyOrg/MyProject ❌
git+git@git.myproject.org:MyProject.git ❌
git+git@git.myproject.org:MyOrg/MyProject.git ❌
git+git@git.myproject.org:MyProject@main ❌
git+git@git.myproject.org:MyProject@6338c7a2525e055a05bae1580e4dd189c2feff7b ❌
```

**Comment by goneall** [comment]
@ianling Thanks for reporting the issue.  I'm in the process of creating a new release and should be able to get a fix for this within the next few days.

**Comment by goneall** [comment]
Resolved with PR #54
